src/cpp/shared/scripting/callback.h

```diff
@@ -43,4 +51,26 @@ void ScriptCallback(const char * funcName, Args... args) {
   }
 }
 
+template<typename Res, typename... Args>
+nonstd::optional<Res> ScriptCallbackResult(const char * funcName, Args... args) {
```

Comment:
Do you want to take `args` as a forwarding reference instead of by value?  Then you can `std::forward` them to `ironfist_lua_pushmulti` and not incur copies.  (And that function should probably take its args by forwarding reference too.)

Reply:
Thanks for the suggestion. I don't think we'll benefit from the time savings, but I do think using std::forward will make this code less readable to most of the team.